CanDIG/CanDIGv2
# Bake the `make init_hosts_file` step so it is no longer needed + Add a `make build-all` command

This PR rolls all of the hosts file steps into `extra_hosts` commands on the docker volumes that require it. This means that the `make init-hosts-file` step is no longer used. This PR also includes my `make build-all` command, that should rebuild the entire CanDIG stack from scratch, without needing to go through all of the steps in `docs/install-docker.md`.

**To test**
- There is a new environment variable in your `.env` file, called [LOCAL_IP_ADDR](https://github.com/CanDIG/CanDIGv2/pull/203/files#diff-94bd2ecddc422d4fa0e5cae5238156d394349819728e1d26f1f04f8168618e26R15). You can set this to your local IP address, or the script will try to automatically find the correct IP to use.
- This requires you to remove any edits to your hosts file that you may have set up (i.e. any `docker.localhost` entries in `/etc/hosts`).
- If you want to run the entire new stack, run `make clean-all`, followed by `make init-conda`, `conda activate candig`, `pip install -U -r etc/venv/requirements.txt`, `make build-all`.
- If you just want to test this in the context of authx, run `make clean-authx` followed by `make init-authx`.
- **NB: The `make init-hosts-file` step is no longer necessary and has been removed**
